src/plugins/share/README.mdx

```diff
@@ -0,0 +1,243 @@
+
+  public readonly getLocation = async ({ name }: HelloLocatorParams) => {
+    return {
+      app: 'sampleAppId',
+      path: `/hello?name=${encodeURIComponent(name)}`,
+      state: {},
+    };
+  };
+}
+```
+
+Once your locator has been defined, you need to register it in the central locator
+registry with the `share` plugin:
+
+```ts
+const locator = plugins.share.url.locators.create(new HelloLocatorDefinition());
+```
+
+You can then return the `locator` reference from your plugin contract fro consumption
+by other plugins:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const locator = plugins.share.url.locators.create(new HelloLocatorDefinition());
+
+    return {
+      locator,
+    };
+  }
+}
+```
+
+### Using locator of another app
+
+First you need to get access of the locator you want to use. Best way to do it is
+by acquiring it from the plugin contract of the app you are interested to navigate
+to. For example, you can get hold of the Dashboard app locator from the `dashboard`
+plugin contract:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const dashboardLocator = plugins.dashboard.locator;
+  }
+}
+```
+
+Alternatively, you can also acquire any locator from the central registry in
+the `share` plugin using the ID of the locator. This is useful, if you run into
+circular dependency issues, however, it is better to acquire the locator directly
+from the plugin contract as above, which makes dependencies explicit.
+
+You can use the `.get()` method on the locators service to acquire the locator
+from the central registry:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const dashboardLocator = plugins.share.url.locators.get('DASHBOARD_APP_LOCATOR');
+  }
+}
+```
+
+Once have have a reference to the desired locator object, you can use that locator
+to navigate to various pages in Kibana or use it to build a URL string.
+
+The best way to use the locator is to use its `.navigateSync()` method to navigate
+in SPA way (without page reload):
+
+```ts
+dashboardLocator.navigateSync({ dashboardId: '123' });
+```
+
+The above will immediately navigate to the Dashboard app, to the dashboard with
+ID `123`.
+
+Another useful feature of locators is their ability to format a URL string of the
+destination. The best way to do it is using the `.getRedirectUrl()` method
+of locators.
+
+```ts
+const url = dashboardLocator.getRedirectUrl({ dashboardId: '123' });
+```
+
+The above will generate a *redirect* URL, which also preserves the serializable
+*location state* object, which will be passed to the destination app.
+
+Alternatively, for compatibility with old URL generators, you can also use the
+`.getUrl()` method of locators to generate just the final app destination URL
+string without the location state object.
+
+```ts
+const url = await dashboardLocator.getUrl({ dashboardId: '123' });
+```
+
+Note, in this process you lose the location state object, which effectively
```

Comment:
put this in a "warning" block as well, its worth highlighting this is a legacy thing and should not be used in the new code.